bin/proving-service/src/proxy/worker.rs

```diff
@@ -32,7 +32,7 @@ const MPS_PROXY_VERSION: &str = env!("CARGO_PKG_VERSION");
 #[derive(Debug, Clone)]
 pub struct Worker {
     backend: Backend,
-    status_client: StatusApiClient<Channel>,
+    status_client: Option<StatusApiClient<Channel>>,
```

Comment:
Would it make sense to create a simple wrapper struct (e.g., `StatusClient`) which would handle status checking (and reconnecting) logic internally? It could look something like this:
```rust
struct StatusClient(Option<StatusApiClient<Channel>>);

impl StatusClient {

    /// Same as current create_status_client()
    pub async fn new(
        address: String,
        connection_timeout: Duration,
        total_timeout: Duration,
    ) -> Result<Self, ProvingServiceError> {
        ...
    }

    /// Handles status fetching and reconnect logic.
    pub async fn check_status(&self, current_status: WorkerHealthStatus) -> WorkerHealthStatus {

    }
}
```